4/Step 1/Shipment.ts

```diff
@@ -0,0 +1,53 @@
+import {Shipper} from "../Step 2/Shipper";
+
+export interface ShipmentInfo {
+    ShipmentID: number;
+    Weight: number;
+    FromAddress: string;
+    FromZipCode: string;
+    ToAddress: string;
+    ToZipCode: string;
+}
+
+const PRICE_PER_OUNCE = 0.39;
+
+export class Shipment {
+    public static id = 0;
+    public static shipment: Shipment;
+
+    public weight: number;
+    public fromAddress: string;
+    public fromZipCode: string;
+    public toAddress: string;
+    public toZipCode: string;
+    public shipmentID: number;
+    public shipper: Shipper;
+
+    constructor(shipment: ShipmentInfo) {
+        this.shipmentID = shipment.ShipmentID === 0 ? this.generateShipmentId() : shipment.ShipmentID;
+        this.weight = shipment.Weight;
+        this.fromAddress = shipment.FromAddress;
+        this.fromZipCode = shipment.FromZipCode;
+        this.toAddress = shipment.ToAddress;
+        this.toZipCode = shipment.ToZipCode;
+
+        this.shipper = Shipper.getInstance(this.fromZipCode);
+    }
+
+    public static getInstance(shipmentInfo: ShipmentInfo): Shipment {
```

Comment:
This specific class is not a good candidate for `Singletone` pattern. By doing this you allow only one instance in runtime but you don't need that. What if, you want to create another shipment with different input values? You will always get the first instance. `Singletone` is mostly used for `Client` classes. For example a class that creates a DB connection and returns all data access methods. For this case you need only single instance to not initialize DB connection every time. Check other creational patterns.

Reply:
ok, I agree but what did they mean for this schema? 
![image](https://github.com/varchenkojulia1/software-designs-and-algorithms/assets/72207362/b5eb5a37-47e2-429a-b8b2-28913ccae22c)
for getInstance function? Just return new Shipment()? Do we need a separate function if no conditions?

Reply:
You don't have to do exactly as in the schema.
![image](https://github.com/varchenkojulia1/software-designs-and-algorithms/assets/37273235/9b2d30d2-4382-4222-9892-bdd3a1272d52)